### PR TITLE
fix: Release41 fixes from NRK

### PR DESCRIPTION
**Pull request by ianshade**
Mainly it brings some critical fixes regarding threadedclass.

### CODE REVIEW

File: meteor/server/api/integration/rabbitMQ.ts [hunk 1]
```diff
@@ -248,7 +254,7 @@ class ChannelManager extends Manager<AMQP.ConfirmChannel> {
 				// Put the message back on the queue:
 				this.outgoingQueue.unshift(messageToSend)
 			} else {
-				logger.debug('RabbitMQ: message sent, waiting for ok...')
+				logger.debug('AMQP: message sent, waiting for ok...')
```
Comment (KvelaGorrrrnio) [inline meteor/server/api/integration/rabbitMQ.ts:257]: Do we now what the message broker is used for?

File: packages/job-worker/src/workers/events/parent.ts [hunk 2]
```diff
@@ -6,6 +6,10 @@ import { getEventsQueueName } from '@sofie-automation/corelib/dist/worker/events
 import { Promisify, threadedClass, ThreadedClassManager } from 'threadedclass'
 import { FastTrackTimelineFunc, LogLineWithSourceFunc } from '../../main'
 
+const FREEZE_LIMIT = 1000 // how long to wait for a response to a Ping
+const RESTART_TIMEOUT = 10000 // how long to wait for a restart to complete before throwing an error
+const KILL_TIMOUT = 10000 // how long to wait for a thread to terminate before throwing an error
```
Comment (KvelaGorrrrnio) [inline packages/job-worker/src/workers/events/parent.ts:11]:
```suggestion
const KILL_TIMEOUT = 10000 // how long to wait for a thread to terminate before throwing an error
```

File: packages/job-worker/src/workers/events/parent.ts [hunk 3]
```diff
@@ -32,6 +36,9 @@ export class EventsWorkerParent extends WorkerParentBase {
 			{
 				instanceName: `Events: ${baseOptions.studioId}`,
 				autoRestart: true,
+				freezeLimit: FREEZE_LIMIT,
+				restartTimeout: RESTART_TIMEOUT,
+				killTimeout: KILL_TIMOUT,
```
Comment (KvelaGorrrrnio) [inline packages/job-worker/src/workers/events/parent.ts:41]:
```suggestion
				killTimeout: KILL_TIMEOUT,
```

File: meteor/server/worker/worker.ts [hunk 4]
```diff
@@ -26,6 +26,10 @@ import * as path from 'path'
 import { LogEntry } from 'winston'
 import { initializeWorkerStatus, setWorkerStatus } from './workerStatus'
 
+const FREEZE_LIMIT = 1000 // how long to wait for a response to a Ping
+const RESTART_TIMEOUT = 30000 // how long to wait for a restart to complete before throwing an error
+const KILL_TIMOUT = 30000 // how long to wait for a thread to terminate before throwing an error
```
Comment (KvelaGorrrrnio) [inline meteor/server/worker/worker.ts:31]:
```suggestion
const KILL_TIMEOUT = 30000 // how long to wait for a thread to terminate before throwing an error
```

File: meteor/server/worker/worker.ts [hunk 5]
```diff
@@ -248,10 +252,20 @@ Meteor.startup(() => {
 			[workerId, jobFinished, interruptJobStream, getNextJob, queueJobWithoutResult, logLine, fastTrackTimeline],
 			{
 				autoRestart: true,
+				freezeLimit: FREEZE_LIMIT,
+				restartTimeout: RESTART_TIMEOUT,
+				killTimeout: KILL_TIMOUT,
```
Comment (KvelaGorrrrnio) [inline meteor/server/worker/worker.ts:257]:
```suggestion
				killTimeout: KILL_TIMEOUT,
```

File: packages/job-worker/src/workers/ingest/parent.ts [hunk 6]
```diff
@@ -6,6 +6,10 @@ import { getIngestQueueName } from '@sofie-automation/corelib/dist/worker/ingest
 import { Promisify, threadedClass, ThreadedClassManager } from 'threadedclass'
 import { FastTrackTimelineFunc, LogLineWithSourceFunc } from '../../main'
 
+const FREEZE_LIMIT = 10000 // how long to wait for a response to a Ping
+const RESTART_TIMEOUT = 10000 // how long to wait for a restart to complete before throwing an error
+const KILL_TIMOUT = 10000 // how long to wait for a thread to terminate before throwing an error
```
Comment (KvelaGorrrrnio) [inline packages/job-worker/src/workers/ingest/parent.ts:11]:
```suggestion
const KILL_TIMEOUT = 10000 // how long to wait for a thread to terminate before throwing an error
```

File: packages/job-worker/src/workers/ingest/parent.ts [hunk 7]
```diff
@@ -33,6 +37,9 @@ export class IngestWorkerParent extends WorkerParentBase {
 			{
 				instanceName: `Ingest: ${baseOptions.studioId}`,
 				autoRestart: true,
+				freezeLimit: FREEZE_LIMIT,
+				restartTimeout: RESTART_TIMEOUT,
+				killTimeout: KILL_TIMOUT,
```
Comment (KvelaGorrrrnio) [inline packages/job-worker/src/workers/ingest/parent.ts:42]:
```suggestion
				killTimeout: KILL_TIMEOUT,
```

File: packages/job-worker/src/workers/studio/parent.ts [hunk 8]
```diff
@@ -6,6 +6,10 @@ import { AnyLockEvent } from '../locks'
 import { Promisify, threadedClass, ThreadedClassManager } from 'threadedclass'
 import { FastTrackTimelineFunc, LogLineWithSourceFunc } from '../../main'
 
+const FREEZE_LIMIT = 2500 // how long to wait for a response to a Ping
+const RESTART_TIMEOUT = 10000 // how long to wait for a restart to complete before throwing an error
+const KILL_TIMOUT = 10000 // how long to wait for a thread to terminate before throwing an error
```
Comment (KvelaGorrrrnio) [inline packages/job-worker/src/workers/studio/parent.ts:11]:
```suggestion
const KILL_TIMEOUT = 10000 // how long to wait for a thread to terminate before throwing an error
```

File: packages/job-worker/src/workers/studio/parent.ts [hunk 9]
```diff
@@ -32,6 +36,9 @@ export class StudioWorkerParent extends WorkerParentBase {
 			{
 				instanceName: `Studio: ${baseOptions.studioId}`,
 				autoRestart: true,
+				freezeLimit: FREEZE_LIMIT,
+				restartTimeout: RESTART_TIMEOUT,
+				killTimeout: KILL_TIMOUT,
```
Comment (KvelaGorrrrnio) [inline packages/job-worker/src/workers/studio/parent.ts:41]:
```suggestion
				killTimeout: KILL_TIMEOUT,
```

**Review by KvelaGorrrrnio — APPROVED** [review]
Added a bunch of suggestions for a typo. It might be cleaner/better to just find-replace in a single commit, than to use the suggestions.. 🙃

File: meteor/server/api/integration/rabbitMQ.ts (hunk 1 above)
Comment (LindvedKrvang) [inline meteor/server/api/integration/rabbitMQ.ts:257]: @jstarpl FYI: This file should be renamed. It's called 'RabbitMQ', but it's using an 'AMQP' integration.
If you are the wrong person to tag, please let me know :)

File: meteor/server/api/userActions.ts [hunk 10]
```diff
@@ -559,6 +559,23 @@ class ServerUserActionAPI
 			}
 		)
 	}
+	async DEBUG_crashStudioWorker(userEvent: string, rundownPlaylistId: RundownPlaylistId) {
```
Comment (LindvedKrvang) [inline meteor/server/api/userActions.ts:562]: What is this used for?

**Review by LindvedKrvang — APPROVED** [review]

File: meteor/server/api/integration/rabbitMQ.ts (hunk 1 above)
Comment (ianshade) [reply meteor/server/api/integration/rabbitMQ.ts:257]: > AMQP is the core protocol for RabbitMQ

File: meteor/server/api/userActions.ts (hunk 10 above)
Comment (ianshade) [reply meteor/server/api/userActions.ts:562]: Looks like it can be used to verify the workers are restarted after they crash.

**Comment by ianshade** [comment]
@KvelaGorrrrnio, turns out the suggestions can be batched into a single commit - that's a neat feature. 😃

**Pull request merged**